libs/designsystem/src/lib/components/calendar/calendar.component.ts

```diff
@@ -57,11 +62,54 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
     this.setActiveMonth(value);
     if (this.hasDateChanged(value, this._selectedDate)) {
       this.onSelectedDateChange(value);
-      this._selectedDate = value;
-      this.dateChange.emit(this._selectedDate);
+      this._selectedDate = this.normalizeDate(value);
+
+      if (!this._selectedDate || this.timezone === 'local') {
```

Comment:
Not very readable - effectively checking for whether this._selectedDate has a value belongs to the `else` statement.
Consider moving there or only emitting if there is a date at all... (Can it be not selected??)

Reply:
Moved the logic so that we now only emit events when the selectedDate changed due to user interaction. This also removed the need for the guard.